lib/server.js

```diff
@@ -41,123 +209,59 @@ function createServer (options, httpHandler) {
-            resolve(logServerAddress())
-          })
-          // we set it afterwards because listen can throw
-          this[kState].listening = true
-        })
+  const firstArg = args[0]
+  const argsLength = args.length
+  const lastArg = args[argsLength - 1]
+  /* Deal with listen (options) || (handle[, backlog]) */
+  if (typeof firstArg === 'object' && firstArg !== null) {
+    options.backlog = argsLength > 1 ? lastArg : undefined
+    Object.assign(options, firstArg)
+  } else if (typeof firstArg === 'string' && isNaN(firstArg)) {
+    /* Deal with listen (pipe[, backlog]) */
+    options.path = firstArg
+    options.backlog = argsLength > 1 ? lastArg : undefined
+  } else {
+    /* Deal with listen ([port[, host[, backlog]]]) */
+    options.port = argsLength >= 1 && firstArg ? firstArg : 0
+    // This will listen to what localhost is.
+    // It can be 127.0.0.1 or ::1, depending on the operating system.
+    // Fixes https://github.com/fastify/fastify/issues/1022.
+    options.host = argsLength >= 2 && args[1] ? args[1] : 'localhost'
+    options.backlog = argsLength >= 3 ? args[2] : undefined
+  }
 
-        return Promise.race([
-          errEvent, // e.g invalid port range error is always emitted before the server listening
-          listen
-        ])
-      })
-    }
+  return options
+}
 
-    const logServerAddress = () => {
-      let address = server.address()
-      const isUnixSocket = typeof address === 'string'
-      /* istanbul ignore next */
-      if (!isUnixSocket) {
-        if (address.address.indexOf(':') === -1) {
-          address = address.address + ':' + address.port
-        } else {
-          address = '[' + address.address + ']:' + address.port
-        }
-      }
-      /* istanbul ignore next */
-      address = (isUnixSocket ? '' : ('http' + (this[kOptions].https ? 's' : '') + '://')) + address
-      this.log.info('Server listening at ' + address)
-      return address
+function logServerAddress (server) {
+  let address = server.address()
+  const isUnixSocket = typeof address === 'string'
+  /* istanbul ignore next */
+  if (!isUnixSocket) {
+    if (address.address.indexOf(':') === -1) {
+      address = address.address + ':' + address.port
+    } else {
+      address = '[' + address.address + ']:' + address.port
     }
+  }
+  /* istanbul ignore next */
+  address = (isUnixSocket ? '' : ('http' + (this[kOptions].https ? 's' : '') + '://')) + address
 
-    if (cb === undefined) return listenPromise(listenOptions)
-
-    this.ready(err => {
-      if (err != null) return cb(err)
-
-      if (this[kState].listening && this[kState].closing) {
-        return cb(new FST_ERR_REOPENED_CLOSE_SERVER(), null)
-      } else if (this[kState].listening) {
-        return cb(new FST_ERR_REOPENED_SERVER(), null)
-      }
-
-      server.once('error', wrap)
-      server.listen(listenOptions, wrap)
-
-      this[kState].listening = true
-    })
+  if (server.isSecondary !== true) {
```

Comment:
Although this change is majorly designed for development use-case. There should be people out there just calling `.listen(port)` in `production` which should trigger the double binding.

I would recommend logging the listening interface no matter it is primary or secondary. Which let the developer aware they are actually binding two interface in any case.